src/com/palmergames/bukkit/util/SupportUtil.java

```diff
@@ -0,0 +1,185 @@
+package com.palmergames.bukkit.util;
+
+import org.bukkit.Bukkit;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.HashMap;
+
+/**
+ * Allows testing support for known unsupported classes
+ * @author GNosii
+ */
+public class SupportUtil {
+
+	private static final HashMap<String, Support> TESTS = getSupportData();
+
+	/**
+	 * Run tests defined in {@link #getSupportData()}
+	 * @return map with classes detected and their support data
+	 */
+	public static HashMap<String, Support> test() {
+		final HashMap<String, Support> map = new HashMap<>();
+		
+		TESTS.forEach((test, support) -> {
+			if (test.startsWith("platform:")) {
+				if (Bukkit.getName().contains(test.replace("platform:", ""))) {
+					map.put(test, support);
+				}
+			}
+			if (test.startsWith("plugin:")) {
+				if (Bukkit.getPluginManager().getPlugin(test.replace("plugin:", "")) != null) {
+					map.put(test, support);
+				}
+			} else {
+				try {
+					Class.forName(test);
+					map.put(test, support);
+				} catch (final ClassNotFoundException ignored) {}
+			}
+		});
+		
+		return map;
+	}
+
+	/**
+	 * Defines which classes should be tested for, and describe their support level and description.
+	 * @return hard-coded map with the qualified name for the class and the support data.
+	 */
+	private static HashMap<String, Support> getSupportData() {
+		final HashMap<String, Support> map = new HashMap<>();
+		
+		/* 
+		 * To add an class to this list, copy and paste its qualified name:
+		 * map.put("com.palmergames.bukkit.towny.Towny", new Support(SupportType.EXTENSION, "Optional description."));
+		 * 
+		 * If the plugin loads before Towny does, and it's available on Bukkit's PluginManager you can use this syntax:
+		 * map.put("plugin:Towny", new Support(SupportType.EXTENSION)
+		 */
+		
+		map.put("plugin:Questioner", new Support(SupportType.UNNECESSARY, "Towny no longer requires Questioner for questions and you can safely remove this plugin."));
+		map.put("plugin:TownyNameUpdater", new Support(SupportType.UNNECESSARY, "Towny no longer depends on TownyNameUpdater for username changes and you can safely remove this plugin."));
+		
+		map.put("org.geysermc.floodgate.SpigotPlugin", new Support(SupportType.UNSUPPORTED_PLUGIN, "Floodgate is known to cause issues regarding their username format."));
+		map.put("com.earth2me.essentials.economy.vault.VaultEconomyProvider", new Support(SupportType.UNSUPPORTED_ECONOMY, "Essentials Economy is known to reset town/nation balances on rare occasions. Be careful if you're using EssentialsEco."));
```

Comment:
This will cause a message to be shown each time Towny loads a class from a plugin that isn't a softdepend, arbitarily loading classes from other plugins is also not really ideal.

The UNSUPPORTED_ECONOMY one should also only warn if the provider is actually in use, otherwise all servers running essentials will get the message on startup.